### PR TITLE
Ticket 4173 : fix multiple fields found in comments

### PR DESCRIPTION
**Pull request by SophieKSTFC**
See https://github.com/ISISComputingGroup/IBEX/issues/4173 

Added a new parsing function `remove_comments` to remove comments from Db records. This function retains hashtags within strings. This new function removes the issues found where comments that include the word 'field' cause the tests to fail due to 'multiple field' instances. This also now fixes the need to have an @ignore for the IPS.db file which contains comments with the word field in.

New tests added in test_all.db to test for multiple edge cases involving comments and hashtags.

### CODE REVIEW

File: db_parser.py [hunk 1]
```diff
@@ -42,7 +42,31 @@ def _get_props(keyword, text):
 
     return fields
 
-
+def remove_comments(line):
```
Comment (LiamPanchaud) [inline db_parser.py:45]: This function almost works but it has issues if there is more than one `#` inside a string. This means that in the case of a slightly more complex `CALC` field, for example, the string it checks will not the correct. 

e.g. on a string like `field(CALC, "A#B?B:A") #comment`, the comment is correctly filtered out and the `#` inside the calc string is retained. If the string contained another `#` as in `field(CALC, "A#B?B:(A#B?A:C") #comment` the returned string is just `field(CALC, "A`. It's a bit contrived, but it would be good if the function would preserve the whole string

Comment (SophieKSTFC) [reply db_parser.py:45]: Will revise the function to ensure strings with multiple # 's are preserved. This will be implemented in the unit test refactoring as it comes under testing the db parser (see ticket https://github.com/ISISComputingGroup/IBEX/issues/2450 )

**Comment by DominicOram** [comment]
Closing as covered by https://github.com/ISISComputingGroup/IBEX/issues/2450
